### PR TITLE
Fix getURL when no resp (could not connect)

### PR DESCRIPTION
**Pull request by fernandog**
@medariox @ratoaq2 fixes all this issues:

old getURL before lab rewrite return None when error

To reproduce just edit network_timezones.py and change url to something wrong like:
` url = 'https://cdnnnnnnn.pymedusa.com/`
and start medusa

also network timezones should be reverted to 'text'. it was correct. getURL that was wrong

### After this PR:
```
23:52:39 DEBUG::DAILYSEARCHER :: [8ae98dd] Error requesting url https://cdnnnnnnn.pymedusa.com/sb_network_timezones/network_timezones.txt. Error: error hostname 'cdnnnnnnn.pymedusa.com' doesn't match either of 'cdn.pymedusa.com', 'www.cdn.pymedusa.com'
23:52:39 WARNING::DAILYSEARCHER :: [8ae98dd] Updating network timezones failed, this can happen from time to time. URL: https://cdnnnnnnn.pymedusa.com/sb_network_timezones/network_timezones.txt

```

### Before
```
Traceback (most recent call last):
  File "SickBeard.py", line 565, in <module>
    medusa.start()
  File "SickBeard.py", line 373, in start
    network_timezones.update_network_dict()
  File "/home/osmc/SickRage/sickbeard/network_timezones.py", line 47, in update_network_dict
    url_data = helpers.getURL(url, session=helpers.make_session(), returns='response')
  File "/home/osmc/SickRage/sickbeard/helpers.py", line 1345, in getURL
    logger.log(u'Error requesting url {resp.url}. Error: {msg}'.format(resp=resp, msg=ex(e)), logger.DEBUG)
UnboundLocalError: local variable 'resp' referenced before assignment
```

```
23:19:23 DEBUG::MAIN :: [8ae98dd] Error requesting url https://cdnnnnnnn.pymedusa.com/sb_network_timezones/network_timezones.txt. Error: error hostname 'cdnnnnnnn.pymedusa.com' doesn't match either of 'cdn.pymedusa.com', 'www.cdn.pymedusa.com'
Traceback (most recent call last):
  File "SickBeard.py", line 565, in <module>
    medusa.start()
  File "SickBeard.py", line 373, in start
    network_timezones.update_network_dict()
  File "/home/osmc/SickRage/sickbeard/network_timezones.py", line 47, in update_network_dict
    url_data = helpers.getURL(url, session=helpers.make_session(), returns='response')
  File "/home/osmc/SickRage/sickbeard/helpers.py", line 1354, in getURL
    return resp
UnboundLocalError: local variable 'resp' referenced before assignment
```
```
23:24:23 DEBUG::MAIN :: [8ae98dd] Error requesting url https://cdnnnnnnn.pymedusa.com/sb_network_timezones/network_timezones.txt. Error: error hostname 'cdnnnnnnn.pymedusa.com' doesn't match either of 'cdn.pymedusa.com', 'www.cdn.pymedusa.com'
Traceback (most recent call last):
  File "SickBeard.py", line 565, in <module>
    medusa.start()
  File "SickBeard.py", line 373, in start
    network_timezones.update_network_dict()
  File "/home/osmc/SickRage/sickbeard/network_timezones.py", line 48, in update_network_dict
    if not url_data.text:
AttributeError: 'NoneType' object has no attribute 'text'
```

### CODE REVIEW

File: sickbeard/helpers.py [hunk 1]
```diff
@@ -1428,15 +1428,17 @@ def getURL(url, post_data=None, params=None, headers=None,  # pylint:disable=too
...
-            logger.log(u'Connection reset by peer accessing url {resp.url}. Error: {msg}'.format(resp=resp, msg=ex(e)), logger.WARNING)
+            logger.log(u'Connection reset by peer accessing url {url}. Error: {msg}'.format(url=url, msg=ex(e)), logger.WARNING)
         else:
-            logger.log(u'Unknown exception in url {resp.url}. Error: {msg}'.format(resp=resp, msg=ex(e)), logger.ERROR)
+            logger.log(u'Unknown exception in url {url}. Error: {msg}'.format(url=url, msg=ex(e)), logger.ERROR)
             logger.log(traceback.format_exc(), logger.DEBUG)
+        return None
 
-    if not response_type or response_type == u'response':
+    if resp and (not response_type or response_type == u'response'):
```
Comment (ratoaq2) [inline sickbeard/helpers.py]: this `if resp` is not needed. All exceptions are already returning: L1432 and L1439.
You can see the normal flow: L1422 to L1429 the resp is already used (L1426).
After your changes in the exception flow, resp will never be None here.

File: sickbeard/helpers.py [hunk 2]
```diff
@@ -1428,15 +1428,17 @@ def getURL(url, post_data=None, params=None, headers=None,  # pylint:disable=too
                        (url=url, status=resp.status_code, desc=http_code_description(resp.status_code)), logger.DEBUG)
```
Comment (ratoaq2) [inline sickbeard/helpers.py]: we should return None as well inside this 'if' statement

File: sickbeard/network_timezones.py [hunk 3]
```diff
@@ -44,15 +44,15 @@ def update_network_dict():
     """Update timezone information from Medusa repositories"""
 
     url = 'https://cdn.pymedusa.com/sb_network_timezones/network_timezones.txt'
-    url_data = helpers.getURL(url, session=helpers.make_session(), returns='response')
-    if not url_data.text:
+    url_data = helpers.getURL(url, session=helpers.make_session(), returns='text')
```
Comment (medariox) [inline sickbeard/network_timezones.py]: This is not the correct way to return text with requests. See documentation: http://docs.python-requests.org/en/master/user/quickstart/#response-content

**Comment by medariox** [comment]
Just FYI, if we decide to return None, which I don't recommend at this point, we will also have to change most providers get_url https://github.com/pymedusa/SickRage/blob/develop/sickbeard/providers/torrent/html/extratorrent.py#L101

**Comment by fernandog** [comment]
@medariox with @ratoaq2 suggestion i get this:

```
09:27:15 DEBUG::MAIN :: [8ae98dd] Error requesting url https://cdnnnnnnn.pymedusa.com/sb_network_timezones/network_timezones.txt. Error: error hostname 'cdnnnnnnn.pymedusa.com' doesn't match either of 'cdn.pymedusa.com', 'www.cdn.pymedusa.com'
Traceback (most recent call last):
  File "SickBeard.py", line 565, in <module>
    medusa.start()
  File "SickBeard.py", line 373, in start
    network_timezones.update_network_dict()
  File "/home/osmc/SickRage/sickbeard/network_timezones.py", line 48, in update_network_dict
    if not url_data.text:
AttributeError: 'NoneType' object has no attribute 'text'

```

**Comment by fernandog** [comment]
tired of this. Im almost reverting the helpers.py rewrite

File: sickbeard/helpers.py [hunk 4]
```diff
@@ -1408,7 +1413,9 @@ def request_defaults(kwargs):
     return hooks, cookies, verify, proxies
 
 
-def getURL(url, post_data=None, params=None, headers=None,  # pylint:disable=too-many-arguments, too-many-return-statements, too-many-branches, too-many-locals
+def getURL(url, post_data=None, params=None, headers=None,  # pylint:disable=too-many-arguments,
+                                                            # too-many-return-statements, too-many-branches,
+                                                            # too-many-locals
```
Comment (fernandog) [inline sickbeard/helpers.py]: @ratoaq2 is this ok?

**Comment by fernandog** [comment]
Issue from user: https://github.com/pymedusa/SickRage/issues/875

@medariox

**Comment by fernandog** [comment]
@ratoaq2 i got this while running flake in command line:

/sickbeard/providers/__init__.py WARNING: __all__ is defined as a list, this means pydocstyle cannot reliably detect contents of the __all__ variable, because it can be mutated. Change __all__ to be an (immutable) tuple, to remove this warning. Note, pydocstyle uses __all__ to detect which definitions are public, to warn if public definitions are missing docstrings. If __all__ is a (mutable) list, pydocstyle cannot reliably assume its contents. pydocstyle will proceed assuming __all__ is not mutated.

**Comment by fernandog** [comment]
@medariox tested a lot (several manual searches, proper search and etc) and no issues.

File: sickbeard/providers/torrent/json/hd4free.py [hunk 5]
```diff
@@ -102,11 +102,6 @@ def search(self, search_strings, age=0, ep_obj=None):  # pylint: disable=too-man
                     logger.log('No data returned from provider', logger.DEBUG)
                     continue
 
-                # Continue only if at least one release is found
-                if not jdata:
```
Comment (fernandog) [inline sickbeard/providers/torrent/json/hd4free.py]: @medariox why did you removed this check?

Comment (medariox) [inline sickbeard/providers/torrent/json/hd4free.py]: Not needed, either it's continued before or it exists.

**Comment by medariox** [comment]
Approved

**Pull request merged**